bdns/dns.go

```diff
@@ -287,25 +288,61 @@ func isPrivateV4(ip net.IP) bool {
 // provided hostname. This method assumes that the external resolver will chase
 // CNAME/DNAME aliases and return relevant A records.  It will retry requests in
 // the case of temporary network errors. It can return net package,
-// context.Canceled, and context.DeadlineExceeded errors.
+// context.Canceled, and context.DeadlineExceeded errors, all wrapped in the DNSError type.
 func (dnsResolver *DNSResolverImpl) LookupHost(ctx context.Context, hostname string) ([]net.IP, error) {
-	var addrs []net.IP
-	dnsType := dns.TypeA
-	r, err := dnsResolver.exchangeOne(ctx, hostname, dnsType, dnsResolver.aStats)
-	if err != nil {
-		return addrs, &DNSError{dnsType, hostname, err, -1}
-	}
-	if r.Rcode != dns.RcodeSuccess {
-		return nil, &DNSError{dnsType, hostname, nil, r.Rcode}
+	errCh := make(chan error, 2)
+	var respA, respAAAA *dns.Msg
+	var wg sync.WaitGroup
+
+	wg.Add(2)
+	go func() {
+		defer wg.Done()
+		var err error
+		respA, err := dnsResolver.exchangeOne(ctx, hostname, dns.TypeA, dnsResolver.aStats)
+		if err != nil {
+			errCh <- &DNSError{dns.TypeA, hostname, err, -1}
+			return
+		}
+		if respA.Rcode != dns.RcodeSuccess {
+			errCh <- &DNSError{dns.TypeA, hostname, nil, respA.Rcode}
+		}
+	}()
+	go func() {
+		defer wg.Done()
+		var err error
+		respAAAA, err := dnsResolver.exchangeOne(ctx, hostname, dns.TypeAAAA, dnsResolver.aStats)
+		if err != nil {
+			errCh <- &DNSError{dns.TypeAAAA, hostname, err, -1}
+			return
+		}
+		if respAAAA.Rcode != dns.RcodeSuccess {
+			errCh <- &DNSError{dns.TypeAAAA, hostname, nil, respAAAA.Rcode}
+		}
+	}()
+
+	wg.Wait()
```

Comment:
> With this we could also return early since we currently only care about looking either up, not both.

I thought we were supposed to be storing all the looked-up IPs in the ValidationRecord. Would it be OK to just store the ipv4 or just store the ipv6?